.github/workflows/test.yml

```diff
@@ -0,0 +1,28 @@
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  # Test that this action works with AWS credentials provided by the
+  # aws-actions/configure-aws-credentials action using GitHub OIDC.
+  #
+  # See https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect
+  test-oidc:
+    runs-on: ubuntu-latest
+    permissions:
+      deployments: write
+      id-token: write
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: eu-west-2
+          role-to-assume: arn:aws:iam::884096606738:role/ably-sdk-builds-sdk-upload-action
+          role-session-name: "${{ github.run_id }}-${{ github.run_number }}"
+      - uses: actions/checkout@v2
+      - uses: ./
```

Comment:
Minor observation but my mild preference would be to synthesise the `test` folder earlier in this workflow rather than statically checkin the `build-output.txt` file... it's a bit incongruous as 'source' and calling it a 'fixture' is perhaps a little ambitious. 😉

Reply:
Ha fair point 🙂 updated in 153b03a